app/models/katello/glue/pulp/repo.rb

```diff
@@ -777,8 +777,33 @@ def capsule_download_policy
         end
       end
 
+      def distributors_match?(capsule_distributors)
+        generated_distributors = self.generate_distributors.map { |dist| object_to_hash(dist) }
+        capsule_distributors.each do |dist|
+          dist.merge!(dist["config"])
+          dist.delete("config")
+        end
+
+        generated_distributors.any? do |gen_dist|
+          capsule_distributors.any? do |cap_dist|
+            (gen_dist.to_a - cap_dist.to_a).empty?
+          end
+        end
+      end
+
+      def importer_matches?(capsule_importer)
+        generated_importer = object_to_hash(self.generate_importer(true))
```

Comment:
Could you just use .as_json() here instead of the object_to_hash method?

```
2.2.4 :013 > Katello::Repository.first.generate_distributors[0].as_json
 => {"relative_url"=>"Default_Organization/Library/custom/testproduct/tester", "http"=>true, "https"=>true, "auto_publish"=>true, "id"=>"Default_Organization-testproduct-tester", "protected"=>true, "checksum_type"=>"sha256"} 
```